demo/full/scripts/controllers/ContentList.jsx

```diff
@@ -34,33 +39,111 @@ class ContentList extends React.Component {
     const {
       url,
       transport,
-      autoPlay,
       supplementaryImageTracks,
       supplementaryTextTracks,
       textTrackMode,
+      drmInfos,
     } = content;
-    loadVideo({
-      url,
-      transport,
-      autoPlay: !(autoPlay === false),
-      supplementaryImageTracks,
-      supplementaryTextTracks,
-      textTrackMode,
+
+    this.buildKeySystems(drmInfos)
+      .then((keySystems) => {
+        loadVideo({
+          url,
+          transport,
+          autoPlay: !(this.state.autoPlay === false),
+          supplementaryImageTracks,
+          supplementaryTextTracks,
+          textTrackMode,
+          keySystems,
+        });
+      });
+  }
+
+  getServerCertificate(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open("GET", url, true);
+      xhr.responseType = "arraybuffer";
+      xhr.onload = (evt) => {
+        if (xhr.status >= 200 && xhr.status < 300) {
+          const serverCertificate = evt.target.response;
+          resolve(serverCertificate);
+        } else {
+          reject();
+        }
+      };
+      xhr.send();
     });
   }
 
-  loadUrl(url) {
-    const { loadVideo } = this.props;
-    loadVideo({
-      url,
-      transport: this.state.transportType.toLowerCase(),
-      autoPlay: true, // TODO add checkBox
-      // native browser subtitles engine (VTTCue) doesn't render stylized subs
-      // we force HTML textTrackMode to vizualise styles
-      textTrackMode: "html",
+  getLicenseCallback(licenseServerUrlValue) {
+    return (challenge) => {
+      const xhr = new XMLHttpRequest();
+      xhr.responseType = "arraybuffer";
+      xhr.open("POST", licenseServerUrlValue, true);
+      return new Promise((resolve, reject) => {
+        xhr.onload = (evt) => {
+          if (xhr.status >= 200 && xhr.status < 300) {
+            const license = evt.target.response;
+            resolve(license);
+          } else {
+            reject();
+          }
+        },
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove use of this comma operator. ([javascript:S878](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS878))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=312&issues=AWXOg0rxlgbYnohYBvKk&open=AWXOg0rxlgbYnohYBvKk)

[ISSUE_KEY]: AWXOg0rxlgbYnohYBvKk